### PR TITLE
Edit workspace

### PR DESCRIPTION
**Pull request by djino97**

### CODE REVIEW

File: src/GUI/Pages/Messages/Workspace/EditWorkspace.razor [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="dropdown-menu" aria-labelledby="dropdown-menu">
+    <a class="dropdown-item" href="#">Rename workspace</a>
+    <a class="dropdown-item" href="#">Edit image</a>
+    <a class="dropdown-item" href="#">Edit image</a>
```
Comment (BelehovEgor) [inline src/GUI/Pages/Messages/Workspace/EditWorkspace.razor]: why did you write it two times

File: src/GUI/Pages/Project/Projects.razor.cs [hunk 2]
```diff
@@ -13,7 +13,8 @@ namespace LT.DigitalOffice.GUI.Pages.Project
 {
     public partial class Projects
     {
-        private int _takeCount = 5;
+        private int _takeCount;
+        private int _takeCountDefault = 5;
```
Comment (BelehovEgor) [inline src/GUI/Pages/Project/Projects.razor.cs]: maybe const

File: src/GUI/Pages/Messages/Workspace/EditWorkspace.razor (hunk 1 above)
Comment (djino97) [reply src/GUI/Pages/Messages/Workspace/EditWorkspace.razor]: Fixed

File: src/GUI/Pages/Project/Projects.razor.cs (hunk 2 above)
Comment (djino97) [reply src/GUI/Pages/Project/Projects.razor.cs]: Fixed

File: src/GUI/Pages/Messages/CreateWorkspace.razor [hunk 3]
```diff
@@ -46,7 +50,7 @@
     {
         try
         {
-            var response = await _MessageService.CreateWorkspaceAsync(_request);
+            var response = await MessageService.CreateWorkspaceAsync(_request);
```
Comment (Glunky) [inline src/GUI/Pages/Messages/CreateWorkspace.razor:53]: dont use var pls

File: src/GUI/Pages/Messages/CreateWorkspace.razor [hunk 4]
```diff
@@ -63,7 +67,7 @@
             _isSuccessOperation = false;
             _messageUser = $"Something went wrong, please try again later.\nMessage: { ex.Result.Message }";
         }
-        catch (Exception ex)
+        catch (Exception)
```
Comment (Glunky) [inline src/GUI/Pages/Messages/CreateWorkspace.razor:70]: just catch then

File: src/GUI/Pages/Messages/CreateWorkspace.razor [hunk 5]
```diff
@@ -82,13 +86,10 @@
                 List<string> imageData = new List<string>(file.Name.Split("."));
                 image.Extension = string.Join("", ".", imageData.Last());
 
-                imageData.RemoveAt(imageData.Count - 1);
-                image.Name = string.Join(".", imageData);
-
                 var fileContent = new StreamContent(file.OpenReadStream(1024 * 1024 * 15));
```
Comment (Glunky) [inline src/GUI/Pages/Messages/CreateWorkspace.razor:89]: var

File: src/GUI/Pages/Messages/CreateWorkspace.razor [hunk 6]
```diff
@@ -82,13 +86,10 @@
...
                 image.Content = Convert.ToBase64String(await fileContent.ReadAsByteArrayAsync());
             }
-            catch (Exception ex)
```
Comment (Glunky) [inline src/GUI/Pages/Messages/CreateWorkspace.razor:91]: catch

File: src/GUI/Pages/Messages/CreateWorkspace.razor (hunk 5 above)
Comment (Glunky) [inline src/GUI/Pages/Messages/CreateWorkspace.razor:89]: var

File: src/GUI/Pages/Messages/Workspace/EditWorkspaceModal.razor.cs [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                _isSuccessOperation = false;
+                _messageUser = $"Something went wrong, please try again later.\nMessage: { ex.Response }";
+            }
+            catch (Exception)
+            {
+                _isSuccessOperation = false;
+                _messageUser = $"Something went wrong, please try again later.";
+            }
+
+            _requestBody = new();
```
Comment (Glunky) [inline src/GUI/Pages/Messages/Workspace/EditWorkspaceModal.razor.cs:81]: what for?

File: src/GUI/Services/ProjectService.cs [hunk 8]
```diff
@@ -44,9 +44,9 @@ public ProjectService(ISessionStorageService storage, IAuthService authService)
             return projectsResponse;
         }
 
-        public async Task<ProjectInfo> CreateProject(ProjectRequest request)
+        public async Task<OperationResultResponse> CreateProject(ProjectRequest request)
```
Comment (Glunky) [inline src/GUI/Services/ProjectService.cs:47]: fix vars in this file pls

File: src/GUI/Services/ProjectService.cs [hunk 9]
```diff
@@ -76,7 +76,7 @@ public async Task<ProjectInfo> CreateProject(ProjectRequest request)
             return await _projectServiceClient.FindTaskPropertiesAsync(token, name, authorId, projectId, skipCount, takeCount);
         }
 
-        public async Task<ProjectResponse> GetProjectAsync(
+        public async Task<OperationResultResponseProjectResponse> GetProjectAsync(
             Guid projectId, 
             bool includeUsers = false,
```
Comment (Glunky) [inline src/GUI/Services/ProjectService.cs:81]: it false by delfault + it always false and it never changed. is it need?

File: src/GUI/Services/MessageService.cs [hunk 10]
```diff
@@ -49,5 +50,15 @@ public async Task<OperationResultResponse> CreateWorkspaceAsync(CreateWorkspaceR
 
             return await _client.GetWorkspaceAsync(token, workspaceId, includeUsers, includeChannels);
         }
+
+        public async Task<OperationResultResponse> EditWorkspaceAsync(
+            Guid workspaceId,
+            List<PatchWorkspaceDocument> body)
+        {
+            await _refreshToken.RefreshAsync();
```
Comment (Glunky) [inline src/GUI/Services/MessageService.cs:58]: use Task.WhenAll

File: src/GUI/Services/MessageService.cs [hunk 11]
```diff
@@ -49,5 +50,15 @@ public async Task<OperationResultResponse> CreateWorkspaceAsync(CreateWorkspaceR
...
+            var token = await _storage.GetItemAsync<string>(Consts.AccessToken);
```
Comment (Glunky) [inline src/GUI/Services/MessageService.cs:59]: use Task.WhenAll + it copypaste 48-49 str

File: src/GUI/Pages/ProjectTask/TaskWindow/TaskModalWindow.razor.cs [hunk 12]
```diff
@@ -43,7 +43,7 @@ public async Task GetTaskAsync(Guid taskId)
             var project = await _ProjectService.GetProjectAsync(taskResponse.Body.Project.Id, includeUsers: true);
```
Comment (Glunky) [inline src/GUI/Pages/ProjectTask/TaskWindow/TaskModalWindow.razor.cs:44]: fix vars

File: src/GUI/Pages/ProjectTask/TaskWindow/EditTaskWindow.razor.cs [hunk 13]
```diff
@@ -44,15 +44,13 @@ protected override void OnInitialized()
...
             _projects = (await _ProjectService.FindProjects(0, int.MaxValue)).Body.ToList();
-            await GetTaskPropertiesAsync();
-            
+            await GetTaskPropertiesAsync(); 
         }
 
         private void SetValueToPatchDocument(PatchDocumentPath path, object value)
         {
-            PatchDocument patch = _requestBody.FirstOrDefault(X => X.Path == path);
+            PatchDocument patch = _requestBody.FirstOrDefault(x => x.Path == path);
```
Comment (Glunky) [inline src/GUI/Pages/ProjectTask/TaskWindow/EditTaskWindow.razor.cs:53]: what for?

File: src/GUI/Pages/Project/ProjectWindow/ProjectModalWindow.razor.cs [hunk 14]
```diff
@@ -16,7 +16,9 @@ public partial class ProjectModalWindow
 
         public async Task GetProjectAsync(Guid projectId)
         {
-            _project = await _ProjectService.GetProjectAsync(projectId, includeUsers: true, includeFiles: true);
+            var response = await _ProjectService.GetProjectAsync(projectId, includeUsers: true, includeFiles: true);
```
Comment (Glunky) [inline src/GUI/Pages/Project/ProjectWindow/ProjectModalWindow.razor.cs:19]: var

File: src/GUI/Pages/Messages/Workspace/Workspace.razor [hunk 15]
```diff
@@ -65,8 +77,8 @@
         }
 
         var response = await messageService.GetWorkspaceAsync(workspaceId);
```
Comment (Glunky) [inline src/GUI/Pages/Messages/Workspace/Workspace.razor:79]: var

File: src/GUI/Services/MessageService.cs (hunk 11 above)
Comment (djino97) [reply src/GUI/Services/MessageService.cs:59]: Why use Task.WhenAll?

File: src/GUI/Pages/Messages/CreateWorkspace.razor (hunk 3 above)
Comment (Cheephot) [reply src/GUI/Pages/Messages/CreateWorkspace.razor:53]: done

File: src/GUI/Pages/Messages/CreateWorkspace.razor (hunk 5 above)
Comment (Cheephot) [reply src/GUI/Pages/Messages/CreateWorkspace.razor:89]: done

Comment (Cheephot) [reply src/GUI/Pages/Messages/CreateWorkspace.razor:89]: done

**Review by Cheephot — COMMENTED** [review]
123

**Pull request closed without merging**